### PR TITLE
[Channel][Admin] Base currency immutable

### PR DESCRIPTION
**Pull request by tuka217**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | no |
| Related tickets | - |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Behat/Context/Ui/Admin/ManagingChannelsContext.php [hunk 1]
```diff
@@ -483,6 +483,17 @@ public function theTaxCalculationStrategyForTheChannelShouldBe(ChannelInterface
     }
 
     /**
+     * @Then the base currency field should be disabled
+     */
+    public function theBaseCurrencyFieldShouldBeDisabled()
+    {
+        Assert::true(
+            $this->updatePage->isBaseCurrencyDisabled(),
+            'Base currency should be immutable, but it does not.'
```
Comment (venyii) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingChannelsContext.php]: `... but it is not.`

File: src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/AddBaseCurrencySubscriber.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    public function preSetData(FormEvent $event)
+    {
+        $resource = $event->getData();
+        $disabled = false;
+
+        if ($resource instanceof ChannelInterface) {
+            $disabled = null !== $resource->getId();
+        } elseif (null !== $resource) {
+            throw new UnexpectedTypeException($resource, ChannelInterface::class);
+        }
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/AddBaseCurrencySubscriber.php]: What would you say for:

```php
private function getDisabledOption($resource)
{
    if ($resource instanceof ChannelInterface) {
        return null !== $resource->getId();
    }

    if (null === $resource) {
        return false;
    }

    throw new UnexpectedTypeException($resource, ChannelInterface::class);
}
```
? We could avoid this nasty `elseif` and make code more readable.

File: src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/AddBaseCurrencySubscriber.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        $disabled = $this->getDisabledOption($resource);
+
+        $form = $event->getForm();
+        $form->add(
+            'baseCurrency',
+            'sylius_currency_choice',
+            [
+                'label' => 'sylius.form.channel.currency_base',
+                'required' => true,
+                'disabled' => $disabled
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/AddBaseCurrencySubscriber.php]: Missing `,` at the end

File: src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/AddBaseCurrencySubscriber.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            ]
+        );
+    }
+
+    /**
+     * @param mixed $resource
+     *
+     * @return bool
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/AddBaseCurrencySubscriber.php]: `@throws UnexpectedTypeException `

**Comment by pjedrzejewski** [comment]
Thank you Ania!

**Pull request merged**